packages/block-library/src/block/edit.js

```diff
@@ -52,79 +60,81 @@ class ReusableBlockEdit extends Component {
 		}
 	}
 
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.reusableBlock !== this.props.reusableBlock && this.state.title === null ) {
+			this.setState( {
+				title: this.props.reusableBlock.title,
+				blocks: parse( this.props.reusableBlock.content ),
+			} );
+		}
+	}
+
 	startEditing() {
 		const { reusableBlock } = this.props;
-
 		this.setState( {
 			isEditing: true,
 			title: reusableBlock.title,
-			changedAttributes: {},
+			blocks: parse( reusableBlock.content ),
 		} );
 	}
 
 	stopEditing() {
 		this.setState( {
 			isEditing: false,
 			title: null,
-			changedAttributes: null,
+			blocks: [],
 		} );
 	}
 
-	setAttributes( attributes ) {
-		this.setState( ( prevState ) => {
-			if ( prevState.changedAttributes !== null ) {
-				return { changedAttributes: { ...prevState.changedAttributes, ...attributes } };
-			}
-		} );
+	setBlocks( blocks ) {
+		this.setState( { blocks } );
 	}
 
 	setTitle( title ) {
 		this.setState( { title } );
 	}
 
 	save() {
-		const { reusableBlock, onUpdateTitle, updateAttributes, block, onSave } = this.props;
-		const { title, changedAttributes } = this.state;
-
-		if ( title !== reusableBlock.title ) {
-			onUpdateTitle( title );
-		}
-
-		updateAttributes( block.clientId, changedAttributes );
+		const { onChange, onSave } = this.props;
+		const { blocks, title } = this.state;
+		const content = serialize( blocks );
+		onChange( { title, content } );
 		onSave();
 
 		this.stopEditing();
 	}
 
 	render() {
-		const { isSelected, reusableBlock, block, isFetching, isSaving, canUpdateBlock } = this.props;
-		const { isEditing, title, changedAttributes } = this.state;
+		const { isSelected, reusableBlock, isFetching, isSaving, canUpdateBlock, settings } = this.props;
+		const { isEditing, title, blocks } = this.state;
 
 		if ( ! reusableBlock && isFetching ) {
 			return <Placeholder><Spinner /></Placeholder>;
 		}
 
-		if ( ! reusableBlock || ! block ) {
+		if ( ! reusableBlock ) {
 			return <Placeholder>{ __( 'Block has been deleted or is unavailable.' ) }</Placeholder>;
 		}
 
 		let element = (
-			<BlockEdit
-				{ ...this.props }
-				isSelected={ isEditing && isSelected }
-				clientId={ block.clientId }
-				name={ block.name }
-				attributes={ { ...block.attributes, ...changedAttributes } }
-				setAttributes={ isEditing ? this.setAttributes : noop }
-			/>
+			<BlockEditorProvider
+				settings={ settings }
+				value={ blocks }
+				onChange={ this.setBlocks }
+				onInput={ this.setBlocks }
+			>
+				<WritingFlow>
+					<BlockList />
```

Comment:
I think we need to add ObserveTyping between WritingFlow and BlockList, if I'm in a paragraph inside reusable blocks and I start typing the toolbar never disappears.